src/wp-includes/class-wp-scripts.php

```diff
@@ -295,14 +295,39 @@ public function do_item( $handle, $group = false ) {
 		}
 
 		$before_handle = $this->print_inline_script( $handle, 'before', false );
-		$after_handle  = $this->print_inline_script( $handle, 'after', false );
 
 		if ( $before_handle ) {
 			$before_handle = sprintf( "<script%s id='%s-js-before'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $before_handle );
 		}
 
-		if ( $after_handle ) {
-			$after_handle = sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_handle );
+		$strategy = $this->get_eligible_loading_strategy( $handle );
+
+		$after_handle = '';
+
+		// Eligible loading strategies will only be 'async', 'defer', or ''.
+		if ( '' === $strategy ) {
+			$after_handle = $this->print_inline_script( $handle, 'after', false );
+
+			if ( $after_handle ) {
+				$after_handle = sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_handle );
+			}
+		} else {
+			$after_standalone_handle = $this->print_inline_script( $handle, 'after-standalone', false );
+
+			if ( $after_standalone_handle ) {
+				$after_handle .= sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_standalone_handle );
+			}
+
+			$after_non_standalone_handle = $this->print_inline_script( $handle, 'after-non-standalone', false );
+
+			if ( $after_non_standalone_handle ) {
+				$after_handle .= sprintf(
+					'<script type=\'text/template\' id=\'%1$s-js-after\' data-wp-executes-after=\'%1$s\'>%3$s%2$s%3$s</script>%3$s',
```

Comment:
```suggestion
					"<script type='text/template' id='%s-js-after' data-wp-executes-after='%s'>\n%s\n</script>\n",
```
@joemcgill  if we want `\n` we will have to bring back double quotes enclosure like in this preview. Down side is we can't use numbered placeholders  `%1$s` as, in double quotes `$s` is taken as a variable which ends up in error. Other issue is we need `handle` name at 2 places so we will have to repeat them.